### PR TITLE
azurerm_logic_app_workflow - support for integration_service_environment_id

### PR DESCRIPTION
**Pull request by favoretti**
Separate work done by @jackofallops on #8315 to unblock workflow ISE
property, while integration account is waiting on upstream fix.

### CODE REVIEW

**Comment by favoretti** [comment]
@jackofallops Since these are all your changes verbatim, I would appreciate it a lot if you could give it a glance and maybe ship with 2.28? Thanks!

File: azurerm/internal/services/logic/validate/logic.go [hunk 1]
```diff
@@ -38,6 +38,20 @@ func IntegrationServiceEnvironmentName() schema.SchemaValidateFunc {
...
 
+func IntegrationServiceEnvironmentID(i interface{}, k string) (warnings []string, errors []error) {
+	v, ok := i.(string)
+	if !ok {
+		errors = append(errors, fmt.Errorf("expected type of %q to be string", k))
+		return warnings, errors
+	}
+
+	if _, err := parse.IntegrationServiceEnvironmentID(v); err != nil {
+		errors = append(errors, fmt.Errorf("con not parse %q as an Integration Service Environment ID: %+v", k, err))
```
Comment (tombuildsstuff) [inline azurerm/internal/services/logic/validate/logic.go:49]:
```suggestion
		errors = append(errors, fmt.Errorf("cannot parse %q as an Integration Service Environment ID: %+v", k, err))
```

File: azurerm/internal/services/logic/logic_app_workflow_resource.go [hunk 2]
```diff
@@ -290,6 +303,10 @@ func resourceArmLogicAppWorkflowRead(d *schema.ResourceData, meta interface{}) e
 			}
 		}
 
+		if props.IntegrationServiceEnvironment != nil && props.IntegrationServiceEnvironment.ID != nil {
+			d.Set("integration_service_environment_id", props.IntegrationServiceEnvironment.ID)
+		}
```
Comment (tombuildsstuff) [inline azurerm/internal/services/logic/logic_app_workflow_resource.go:308]: we need to ensure this is always set into the state - so can we make this:

```suggestion
		integrationServiceEnvironmentId := ""
		if props.IntegrationServiceEnvironment != nil && props.IntegrationServiceEnvironment.ID != nil {
 			integrationServiceEnvironmentId := *props.IntegrationServiceEnvironment.ID
 		}
 		d.Set("integration_service_environment_id", integrationServiceEnvironmentId)
```

File: azurerm/internal/services/logic/logic_app_workflow_resource.go [hunk 3]
```diff
@@ -290,6 +303,10 @@ func resourceArmLogicAppWorkflowRead(d *schema.ResourceData, meta interface{}) e
...
+
 		if props.IntegrationAccount != nil && props.IntegrationAccount.ID != nil {
 			d.Set("logic_app_integration_account_id", props.IntegrationAccount.ID)
 		}
```
Comment (tombuildsstuff) [inline azurerm/internal/services/logic/logic_app_workflow_resource.go:312]: I realise this isn't your change - but as we're changing the item above, could we fix this too:

```suggestion
		integrationAccountId := ""
		if props.IntegrationAccount != nil && props.IntegrationAccount.ID != nil {
 			integrationAccountId := *props.IntegrationAccount.ID
 		}
 		d.Set("logic_app_integration_account_id", integrationAccountId)
```

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
hey @favoretti 

Thanks for this PR :)

Taking a look through this mostly LGTM - if we can fix up the minor comments then this should otherwise be good to merge 👍 

Thanks!

**Review by tombuildsstuff — APPROVED** [review]
LGTM 👍

**Comment by tombuildsstuff** [comment]
@favoretti running the tests for this - all of the integration service environment tests are failing with:

```
------- Stdout: -------
=== RUN   TestAccAzureRMIntegrationServiceEnvironment_update
=== PAUSE TestAccAzureRMIntegrationServiceEnvironment_update
=== CONT  TestAccAzureRMIntegrationServiceEnvironment_update
TestAccAzureRMIntegrationServiceEnvironment_update: testing.go:684: Step 0 error: Expected a non-empty plan, but got an empty plan!
--- FAIL: TestAccAzureRMIntegrationServiceEnvironment_update (10565.84s)
FAIL
```

Which I believe is due to these: https://github.com/terraform-providers/terraform-provider-azurerm/blob/07b3926964c9792419206c2dac6af1ab0b60dfbf/azurerm/internal/services/logic/tests/integration_service_environment_resource_test.go#L134 - would you be able to move those from the ISE tests and we should be able to re-run those?

Thanks!

**Comment by favoretti** [comment]
Hmm, I'm not sure I follow. Why would that start failing all of a sudden? And by "move" you mean make it accept empty plans? Would not that break anything else?

**Comment by favoretti** [comment]
@tombuildsstuff Pushed a commit removing those.. I'll run acceptance tests here locally as well to see if it works.

**Comment by favoretti** [comment]
@tombuildsstuff I ran acceptance tests locally, one failed, but really due to ISE's being slow, the rest passed:
```
==> Checking that code complies with gofmt requirements...
==> Checking that Custom Timeouts are used...
==> Checking that acceptance test packages are used...
TF_ACC=1 go test -v ./azurerm/internal/services/logic/tests/ -run=TestAccAzureRMIntegrationServiceEnvironment -timeout 600m -ldflags="-X=github.com/terraform-providers/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccAzureRMIntegrationServiceEnvironment_basic
=== PAUSE TestAccAzureRMIntegrationServiceEnvironment_basic
=== RUN   TestAccAzureRMIntegrationServiceEnvironment_complete
=== PAUSE TestAccAzureRMIntegrationServiceEnvironment_complete
=== RUN   TestAccAzureRMIntegrationServiceEnvironment_developer
=== PAUSE TestAccAzureRMIntegrationServiceEnvironment_developer
=== RUN   TestAccAzureRMIntegrationServiceEnvironment_update
=== PAUSE TestAccAzureRMIntegrationServiceEnvironment_update
=== RUN   TestAccAzureRMIntegrationServiceEnvironment_requiresImport
=== PAUSE TestAccAzureRMIntegrationServiceEnvironment_requiresImport
=== CONT  TestAccAzureRMIntegrationServiceEnvironment_basic
=== CONT  TestAccAzureRMIntegrationServiceEnvironment_update
=== CONT  TestAccAzureRMIntegrationServiceEnvironment_requiresImport
=== CONT  TestAccAzureRMIntegrationServiceEnvironment_developer
=== CONT  TestAccAzureRMIntegrationServiceEnvironment_complete
=== CONT  TestAccAzureRMIntegrationServiceEnvironment_update
    testing.go:684: Step 0 error: errors during apply:

        Error: waiting for completion of Integration Service Environment "acctestRG-logic-200924222927093004" (Resource Group "acctestRG-logic-200924222927093004"): Code="InternalServerError" Message="Encountered internal server error. The tracking Id is '7ad6878f-7bac-4790-8ce6-f04a42f4dd65'."

          on /var/folders/_j/kwsxcxl974z5c6hz5lgldrkh0000gn/T/tf-test173134034/main.tf line 56:
          (source code not available)


    testing.go:745: Error destroying resource! WARNING: Dangling resources
        may exist. The full state and error is shown below.

        Error: errors during apply: 4 problems:

        - Error deleting Subnet "isesubnet2" (Virtual Network "acctest-vnet-200924222927093004" / Resource Group "acctestRG-logic-200924222927093004"): network.SubnetsClient#Delete: Failure sending request: StatusCode=400 -- Original Error: Code="InUseSubnetCannotBeDeleted" Message="Subnet isesubnet2 is in use by /subscriptions/f7fd1983-0403-412d-922e-781bd7542ab6/providers/Microsoft.ApiManagement/service?vnetResourceGuid=7ca52618-9eee-42c6-aef6-ed26b93823d5&subnet=isesubnet2&api-version=2016-07-07 and cannot be deleted. In order to delete the subnet, delete all the resources within the subnet. See aka.ms/deletesubnet." Details=[]
        - Error deleting Subnet "isesubnet1" (Virtual Network "acctest-vnet-200924222927093004" / Resource Group "acctestRG-logic-200924222927093004"): network.SubnetsClient#Delete: Failure sending request: StatusCode=400 -- Original Error: Code="InUseSubnetCannotBeDeleted" Message="Subnet isesubnet1 is in use by /subscriptions/a31346ad-8afd-4adb-910b-4865f7f6ca0f/resourceGroups/acctestRG-logic-200924222927093004/providers/Microsoft.Network/virtualNetworks/acctest-vnet-200924222927093004/subnets/isesubnet1/serviceAssociationLinks/MicrosoftLogic-ISE-d84162826e63492b9716b605d4849c6f-o2uwxnn3fxkfo-SAL and cannot be deleted. In order to delete the subnet, delete all the resources within the subnet. See aka.ms/deletesubnet." Details=[]
        - Error deleting Subnet "isesubnet4" (Virtual Network "acctest-vnet-200924222927093004" / Resource Group "acctestRG-logic-200924222927093004"): network.SubnetsClient#Delete: Failure sending request: StatusCode=400 -- Original Error: Code="InUseSubnetCannotBeDeleted" Message="Subnet isesubnet4 is in use by /subscriptions/b434325e-47b1-4184-91dc-da612e81e2f0/resourceGroups/eus2_49916_52118f1ea8de437a895fa53b76a69628/providers/Microsoft.Network/loadBalancers/eus2-49916-163578083/frontendIPConfigurations/frontend and cannot be deleted. In order to delete the subnet, delete all the resources within the subnet. See aka.ms/deletesubnet." Details=[]
        - Error deleting Subnet "isesubnet3" (Virtual Network "acctest-vnet-200924222927093004" / Resource Group "acctestRG-logic-200924222927093004"): network.SubnetsClient#Delete: Failure sending request: StatusCode=400 -- Original Error: Code="InUseSubnetCannotBeDeleted" Message="Subnet isesubnet3 is in use by /subscriptions/f7fd1983-0403-412d-922e-781bd7542ab6/resourceGroups/flowo2uwxnn3fxkfobn00-connectors/providers/Microsoft.Web/hostingEnvironments/flow-o2uwxnn3fxkfo-bn-ase and cannot be deleted. In order to delete the subnet, delete all the resources within the subnet. See aka.ms/deletesubnet." Details=[]

        State: azurerm_resource_group.test:
          ID = /subscriptions/a31346ad-8afd-4adb-910b-4865f7f6ca0f/resourceGroups/acctestRG-logic-200924222927093004
          provider = provider.azurerm
          location = eastus2
          name = acctestRG-logic-200924222927093004
          tags.% = 0
        azurerm_subnet.isesubnet1:
          ID = /subscriptions/a31346ad-8afd-4adb-910b-4865f7f6ca0f/resourceGroups/acctestRG-logic-200924222927093004/providers/Microsoft.Network/virtualNetworks/acctest-vnet-200924222927093004/subnets/isesubnet1
          provider = provider.azurerm
          address_prefix = 10.0.1.0/26
          address_prefixes.# = 1
          address_prefixes.0 = 10.0.1.0/26
          delegation.# = 1
          delegation.0.name = integrationServiceEnvironments
          delegation.0.service_delegation.# = 1
          delegation.0.service_delegation.0.actions.# = 1
          delegation.0.service_delegation.0.actions.0 = Microsoft.Network/virtualNetworks/subnets/action
          delegation.0.service_delegation.0.name = Microsoft.Logic/integrationServiceEnvironments
          enforce_private_link_endpoint_network_policies = false
          enforce_private_link_service_network_policies = false
          name = isesubnet1
          resource_group_name = acctestRG-logic-200924222927093004
          service_endpoints.# = 0
          virtual_network_name = acctest-vnet-200924222927093004
        azurerm_subnet.isesubnet2:
          ID = /subscriptions/a31346ad-8afd-4adb-910b-4865f7f6ca0f/resourceGroups/acctestRG-logic-200924222927093004/providers/Microsoft.Network/virtualNetworks/acctest-vnet-200924222927093004/subnets/isesubnet2
          provider = provider.azurerm
          address_prefix = 10.0.1.64/26
          address_prefixes.# = 1
          address_prefixes.0 = 10.0.1.64/26
          delegation.# = 0
          enforce_private_link_endpoint_network_policies = false
          enforce_private_link_service_network_policies = false
          name = isesubnet2
          resource_group_name = acctestRG-logic-200924222927093004
          service_endpoints.# = 0
          virtual_network_name = acctest-vnet-200924222927093004
        azurerm_subnet.isesubnet3:
          ID = /subscriptions/a31346ad-8afd-4adb-910b-4865f7f6ca0f/resourceGroups/acctestRG-logic-200924222927093004/providers/Microsoft.Network/virtualNetworks/acctest-vnet-200924222927093004/subnets/isesubnet3
          provider = provider.azurerm
          address_prefix = 10.0.1.128/26
          address_prefixes.# = 1
          address_prefixes.0 = 10.0.1.128/26
          delegation.# = 0
          enforce_private_link_endpoint_network_policies = false
          enforce_private_link_service_network_policies = false
          name = isesubnet3
          resource_group_name = acctestRG-logic-200924222927093004
          service_endpoints.# = 0
          virtual_network_name = acctest-vnet-200924222927093004
        azurerm_subnet.isesubnet4:
          ID = /subscriptions/a31346ad-8afd-4adb-910b-4865f7f6ca0f/resourceGroups/acctestRG-logic-200924222927093004/providers/Microsoft.Network/virtualNetworks/acctest-vnet-200924222927093004/subnets/isesubnet4
          provider = provider.azurerm
          address_prefix = 10.0.1.192/26
          address_prefixes.# = 1
          address_prefixes.0 = 10.0.1.192/26
          delegation.# = 0
          enforce_private_link_endpoint_network_policies = false
          enforce_private_link_service_network_policies = false
          name = isesubnet4
          resource_group_name = acctestRG-logic-200924222927093004
          service_endpoints.# = 0
          virtual_network_name = acctest-vnet-200924222927093004
        azurerm_virtual_network.test:
          ID = /subscriptions/a31346ad-8afd-4adb-910b-4865f7f6ca0f/resourceGroups/acctestRG-logic-200924222927093004/providers/Microsoft.Network/virtualNetworks/acctest-vnet-200924222927093004
          provider = provider.azurerm
          address_space.# = 1
          address_space.0 = 10.0.0.0/22
          ddos_protection_plan.# = 0
          dns_servers.# = 0
          guid = 7ca52618-9eee-42c6-aef6-ed26b93823d5
          location = eastus2
          name = acctest-vnet-200924222927093004
          resource_group_name = acctestRG-logic-200924222927093004
          subnet.# = 4
          subnet.1942165044.address_prefix = 10.0.1.0/26
          subnet.1942165044.id = /subscriptions/a31346ad-8afd-4adb-910b-4865f7f6ca0f/resourceGroups/acctestRG-logic-200924222927093004/providers/Microsoft.Network/virtualNetworks/acctest-vnet-200924222927093004/subnets/isesubnet1
          subnet.1942165044.name = isesubnet1
          subnet.1942165044.security_group =
          subnet.2672140647.address_prefix = 10.0.1.192/26
          subnet.2672140647.id = /subscriptions/a31346ad-8afd-4adb-910b-4865f7f6ca0f/resourceGroups/acctestRG-logic-200924222927093004/providers/Microsoft.Network/virtualNetworks/acctest-vnet-200924222927093004/subnets/isesubnet4
          subnet.2672140647.name = isesubnet4
          subnet.2672140647.security_group =
          subnet.3806714578.address_prefix = 10.0.1.128/26
          subnet.3806714578.id = /subscriptions/a31346ad-8afd-4adb-910b-4865f7f6ca0f/resourceGroups/acctestRG-logic-200924222927093004/providers/Microsoft.Network/virtualNetworks/acctest-vnet-200924222927093004/subnets/isesubnet3
          subnet.3806714578.name = isesubnet3
          subnet.3806714578.security_group =
          subnet.4185911648.address_prefix = 10.0.1.64/26
          subnet.4185911648.id = /subscriptions/a31346ad-8afd-4adb-910b-4865f7f6ca0f/resourceGroups/acctestRG-logic-200924222927093004/providers/Microsoft.Network/virtualNetworks/acctest-vnet-200924222927093004/subnets/isesubnet2
          subnet.4185911648.name = isesubnet2
          subnet.4185911648.security_group =
          tags.% = 0

          Dependencies:
            azurerm_resource_group.test
--- FAIL: TestAccAzureRMIntegrationServiceEnvironment_update (5797.11s)
--- PASS: TestAccAzureRMIntegrationServiceEnvironment_developer (9175.94s)
--- PASS: TestAccAzureRMIntegrationServiceEnvironment_complete (10908.32s)
--- PASS: TestAccAzureRMIntegrationServiceEnvironment_requiresImport (12036.89s)
--- PASS: TestAccAzureRMIntegrationServiceEnvironment_basic (12177.99s)
```

Normal tests failed on website-test transient tooling install error. Would you be so kind to retrigger both?

**Comment by tombuildsstuff** [comment]
Tests pass:

Screenshot 2020-09-25 at 16 48 37

**Pull request merged**